src/Fmt_ast.ml

```diff
@@ -3653,9 +3653,13 @@ and fmt_structure c ctx itms =
           fmt_structure_item c ~last:(last && last_grp) (sub_str ~ctx itm)
     )
   in
+  let unix_line_endings =
+    Poly.(Conf.line_endings c.conf (Source.input_filename c.source) = `Unix)
+  in
   hvbox 0
     (list_fl grps (fun ~first ~last grp ->
-         fmt_if (not first) "\n@\n" $ fmt_grp ~last grp ))
+         fmt_if_k (not first) (fmt_or unix_line_endings "\n@\n" "@\n@\n")
+         $ fmt_grp ~last grp ))
```

Comment:
If you have to go to such lengths as this, for such a corner case, I think it would be better to just post-process with sed in the cases where generating windows endings when running on unix. It would be nice to only override the out_newline function, but if that doesn't work, it's going to be too much work and be too invasive. All the other cases seem to have working nice implementations now though.

Reply:
I replaced the last commits with a basic post-processing. I still have an issue with `make test`, looks like the options in attributes are broken (same problem in other PRs as well).